### PR TITLE
Added firbase authentication

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Partially solves #707 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:
- Added Firebase Signup

### CODE REVIEW

File: backend/blueprint/api/__init__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def create_app():
+    app = Flask(__name__, instance_relative_config=True)
+    app.config['BASE_DIR'] = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+    app.config.from_object(os.getenv('APP_CONFIG', default='api.config.config.ProductionConfig'))
+    app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = True
+    app.config.from_object('api.config.mailConfig.MailConfig')
+    cred = credentials.Certificate(os.getenv("FIREBASE_CONFIG"))
+    firebase_admin.initialize_app(cred)
```
Comment (yashLadha) [inline backend/blueprint/api/__init__.py]: Changes Already implemented : https://github.com/fossasia/badgeyay/blob/77fbdd2c132631aff1860836cc090a99b059bf2a/backend/blueprint/api/__init__.py#L11

File: backend/blueprint/api/controllers/registerUser.py [hunk 2]
```diff
@@ -14,20 +14,15 @@ def registerUser():
...
-        data['password'],
-        data['name'],
-        data['email'])
-
     try:
-        newUser.save_to_db()
+        auth.create_user(
+            display_name=data['name'],
+            email=data['email'],
+            phone_number=data['phone_number'],
```
Comment (yashLadha) [inline backend/blueprint/api/controllers/registerUser.py]: Don't copy directly from the documentation. Please see the use cases that are currently present in the system. Is there `photo_url` or `phone_number` present. If not them first add and then update. Please read documentation for reference don't just copy paste the code.

**Comment by yashLadha** [comment]
Change title of the PR. There is a typo there.

File: backend/blueprint/api/controllers/registerUser.py (hunk 2 above)
Comment (yashLadha) [reply backend/blueprint/api/controllers/registerUser.py]: It is also returning a `userRecord` which can be of use later in future so store the necessary details in db. Ref [Link](https://firebase.google.com/docs/reference/admin/python/firebase_admin.auth#create_user)

Comment (gabru-md) [reply backend/blueprint/api/controllers/registerUser.py]: @vaibhavsingh97 please send a PR that has complete work done. And not partial fixes for small tasks.
Moreover please do not copy and paste from docs as Yash pointed. You should atleast look into what you're pasting and if it is even relevant to the context of the project.

**Comment by djmgit** [comment]
@vaibhavsingh97 is this relevant any more. Since you are working on GCP integration and badge data storage, you can consider closing this.

**Comment by vaibhavsingh97** [comment]
@djmgit Yes, We can close this as @ParthS007 already pushed an update for this

**Pull request closed without merging**